File: semester2/5_Fuzzy-Software-Technologies/aics-fuzzy-software-technologies/ticket-backend/src/main/java/aics/domain/fuzzy/dtos/TopsisDataTableDto.java
package aics.domain.fuzzy.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.util.List;

@Data
@Accessors(chain = true)
@RequiredArgsConstructor
@AllArgsConstructor
public class TopsisDataTableDto implements Serializable {
    private List<TopsisDataRowDto> rows;
    private boolean showDpos;
    private boolean showDneg;
    private boolean showScore;
}

File: semester2/5_Fuzzy-Software-Technologies/aics-fuzzy-software-technologies/ticket-frontend/src/modules/fuzzy/dtos/topsis-data-table-dto.ts
import { TopsisDataRowDto } from './topsis-data-row-dto';

export class TopsisDataTableDto {
    rows: TopsisDataRowDto[];
    showDpos: boolean;
    showDneg: boolean;
    showScore: boolean;

    constructor(obj: {
        rows: TopsisDataRowDto[],
        showDpos: boolean,
        showDneg: boolean,
        showScore: boolean,
    }) {
        this.rows = obj.rows;
        this.showDpos = obj.showDpos;
        this.showDneg = obj.showDneg;
        this.showScore = obj.showScore;
    }

    static fromObj(obj: any): TopsisDataTableDto {
        return new TopsisDataTableDto({
            rows: obj.rows ? (obj.rows as any[]).map(row => TopsisDataRowDto.fromObj(row)) : [],
            showDpos: obj.showDpos,
            showDneg: obj.showDneg,
            showScore: obj.showScore,
        })
    }

}
